### PR TITLE
webhooks: Remove build worker

### PR DESCRIPTION
**Pull request by ryanslade**
Remove the code associated with webhook build jobs. 

They are not being used and it is confusing to leave the code around.

The database tables have *not* been removed as this would not be a backwards compatible change. They will be removed in the next release.

Part of https://github.com/sourcegraph/sourcegraph/issues/46562

## Test plan

All tests pass and ran a local instance too to manually double check.

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]
Nice!

**Comment by ryanslade** [comment]
Dry run build here: 
https://buildkite.com/sourcegraph/sourcegraph/builds/193912

If that passes I'm tempted to sneak this into the release so that we can add the database migrations tomorrow after branch cut.

File: schema/site.schema.json [hunk 1]
```diff
@@ -360,7 +360,8 @@
           "description": "Enable webhooks for repo syncing",
           "type": "boolean",
           "default": false,
-          "!go": { "pointer": false }
+          "!go": { "pointer": false },
+          "deprecationMessage": "This setting is deprecated and is ignored. Syncing repositories via webhooks is enabled by default."
```
Comment (mrnugget) [inline schema/site.schema.json:364]: I think you can just remove it now. We now allow additionalKeys in our config, so you can customers who have this in their schema won't get errors but warnings that this is an unknown key.

Do we know of any customers who have enabled this? I can't find any mentions on a docs page, or a Markdown file, or something like that. So if we haven't told a customer explicitly about this chances seem low that they discover this on their own

Comment (ryanslade) [reply schema/site.schema.json:364]: AFAIK nobody is using it. It's always been experimental.

I didn't realise it was safe to remove config now so I'll do that.

**Pull request merged**